src/Compilers/Core/Portable/Operations/IOperation.OperationList.Reversed.cs

```diff
@@ -0,0 +1,111 @@
+            /// </summary>
+            [NonDefaultable]
+            public readonly struct Reversed : IReadOnlyCollection<IOperation>
+            {
+                private readonly Operation _operation;
+
+                internal Reversed(Operation operation)
+                {
+                    _operation = operation;
+                }
+
+                public int Count => _operation.ChildOperationsCount;
+
+                public Enumerator GetEnumerator() => new Enumerator(_operation);
+
+                public ImmutableArray<IOperation> ToImmutableArray()
+                {
+                    Enumerator enumerator = GetEnumerator();
+                    switch (_operation)
+                    {
+                        case NoneOperation { Children: var children }:
+                            return children;
+                        case InvalidOperation { Children: var children }:
+                            return children;
+                        case { ChildOperationsCount: 0 }:
+                            return ImmutableArray<IOperation>.Empty;
+                        default:
+                            var builder = ArrayBuilder<IOperation>.GetInstance(Count);
+                            foreach (var child in this)
+                            {
+                                builder.Add(child);
+                            }
+                            return builder.ToImmutableAndFree();
+                    }
+                }
+
+                IEnumerator<IOperation> IEnumerable<IOperation>.GetEnumerator() => this.GetEnumerator();
+                IEnumerator IEnumerable.GetEnumerator() => this.GetEnumerator();
+
+                /// <summary>
+                /// Implements a reverse-order struct-based enumerator for <see cref="Operation"/> nodes. This type is not hardened
+                /// to <code>default(Enumerator)</code>, and will null reference in these cases. Calling <see cref="Current"/> after
+                /// <see cref="Enumerator.MoveNext"/> has returned false will throw an <see cref="InvalidOperationException"/>.
+                /// </summary>
+                [NonDefaultable]
+                public struct Enumerator : IEnumerator<IOperation>
+                {
+                    private readonly Operation _operation;
+                    private int _currentSlot;
+                    private int _currentIndex;
+
+                    internal Enumerator(Operation operation)
+                    {
+                        _operation = operation;
+                        _currentSlot = int.MaxValue;
+                        _currentIndex = int.MaxValue;
+                    }
+
+                    public IOperation Current
+                    {
+                        get
+                        {
+                            Debug.Assert(_operation != null && _currentSlot is >= 0 and not int.MaxValue && _currentIndex is >= 0 and not int.MaxValue);
```

Comment:
>_currentSlot is >= 0 and not int.MaxValue && _currentIndex is >= 0 and not int.MaxValue [](http://example.com/codeflow?start=63&length=87)

`Current` is a `public` property so we probably cannot assert this.

Reply:
Same comment as on the other `Current` implementation.